go-controller/pkg/ovn/egressgw.go

```diff
@@ -185,6 +181,12 @@ func (oc *Controller) deletePodGWRoutesForNamespace(pod, namespace, node string)
 				// clean up if there are no more routes for this podIP
 				if entry := nsInfo.podExternalRoutes[podIP]; len(entry) == 0 {
 					delete(nsInfo.podExternalRoutes, podIP)
+					// TODO (trozet): use the go bindings here and batch commands
+					// delete the ovn_cluster_router policy if the pod has no more exgws to revert back to normal
+					// default gw behavior
+					if err := oc.delHybridRoutePolicyForPod(net.ParseIP(podIP), node); err != nil {
+						klog.Error(err)
```

Comment:
@trozet the `node` above seems to be the Node on which the External Gateway Pod resided and that pod was deleted from. However, you want that `node` to be the Node on which the podIP resides on, isn't it?

You are walking through each of the Pod in namespace NS and if there are no more external gateways for those pods, then you want to delete the the logical router policie on ovn_cluster_router for that pod.

Reply:
great catch @girishmg ! thanks, pushing new version.